### PR TITLE
Move from `pytorch_lightning` to `lightning`

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Fix the breaking change to `load_from_checkpoint` introduced by `pytorch_lightning==2.1.0`.
- Install the new package name `lightning` instead of `pytorch_lightning`
- Use the new import style `lightning.pytorch` instead of `pytorch_lightning`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: docs/tutorials/advanced_topics/index.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ```{toctree}
 :maxdepth: 1
-howto_pytorch_lightning
+howto_lightning.pytorch
```
Comment (lostella) [inline docs/tutorials/advanced_topics/index.md]: This slipped in I guess?

**Review by lostella — CHANGES_REQUESTED** [review]
One change to revert in the docs I believe, otherwise looks good, thank you @shchur!

File: docs/tutorials/advanced_topics/index.md (hunk 1 above)
Comment (lostella) [inline docs/tutorials/advanced_topics/index.md]:
```suggestion
howto_pytorch_lightning
```

**Review by lostella — APPROVED** [review]

**Comment by caseytomlin** [comment]
Hi, maybe I'm doing something else wrong, but this new import syntax seems to not work with `pytorch_lightning<2.0.0`, but the [requirements](https://github.com/shchur/gluonts/blob/a8a5ec0602286f5f842bbc89adfdf838a95c2492/requirements/requirements-pytorch.txt) allow otherwise. is it just me?

image

**Comment by shchur** [comment]
Thanks @caseytomlin, we have released v0.13.7 that uses the old import style and therefore is compatible with the correct version ranges.

We will update the version ranges in v0.14.0, where the the new import style will be introduced.

**Pull request merged**